spirv_glsl.cpp

```diff
@@ -5691,17 +5672,22 @@ string CompilerGLSL::legacy_tex_op(const std::string &op, const SPIRType &imgtyp
 		break;
 	}
 
-	bool use_explicit_lod = check_explicit_lod_allowed(lod);
-
-	if (op == "textureLod" || op == "textureProjLod" || op == "textureGrad" || op == "textureProjGrad")
+	// In legacy GLSL, an extension is required for textureLod in the fragment
+	// shader or textureGrad anywhere.
+	bool legacy_lod_ext = false;
+	auto &execution = get_entry_point();
+	if (op == "textureGrad" || op == "textureProjGrad" ||
+		((op == "textureLod" || op == "textureProjLod") && execution.model != ExecutionModelVertex))
```

Comment:
Formatting, `((op == "textureLod` should align with `op == "textureGrad` using spaces.